vador/src/main/java/com/salesforce/vador/execution/strategies/AnnotationProcessor.kt

```diff
@@ -24,27 +35,120 @@ object AnnotationProcessorBase {
+                    none,
+                  )
+                }
+              liftToEtr<FailureT, ValidatableT>(validator, none)
+            } else {
+              throw IllegalArgumentException(
+                "Provided Annotation is not supported. Please use supported annotations or custom annotations with valid instance. For more info please check official documentation.",
               )
             }
-          liftToEtr<FailureT, ValidatableT>(validator, none)
-        } else {
-          throw IllegalArgumentException(
-            "Provided Annotation is not supported. Please use supported annotations or custom annotations with valid instance. For more info please check official documentation."
-          )
+          }
+          "Negative" -> {
+            val annotation = field.getAnnotation(Negative::class.java)
+            val failureKey = annotation.failureKey
+            val negativeValidator =
+              NegativeValidator::class.java.getDeclaredConstructor().newInstance()
+                as ValidatorAnnotation1<Any, FailureT?>
+            validator =
+              Validator<ValidatableT?, FailureT?> {
+                negativeValidator.validate(
+                  field,
+                  FieldUtils.readField(field, bean, true) as Any,
+                  map?.get(failureKey),
+                  none,
+                )
+              }
+            liftToEtr<FailureT, ValidatableT>(validator, none)
+          }
+          "NonNegative" -> {
+            val annotation = field.getAnnotation(NonNegative::class.java)
+            val failureKey = annotation.failureKey
+            val nonNegativeValidator =
+              NonNegativeValidator::class.java.getDeclaredConstructor().newInstance()
+                as ValidatorAnnotation1<Any, FailureT?>
+            validator =
+              Validator<ValidatableT?, FailureT?> {
+                nonNegativeValidator.validate(
+                  field,
+                  FieldUtils.readField(field, bean, true) as Any,
+                  map?.get(failureKey),
+                  none,
+                )
+              }
+            liftToEtr<FailureT, ValidatableT>(validator, none)
+          }
+          "Positive" -> {
+            val annotation = field.getAnnotation(Positive::class.java)
+            val failureKey = annotation.failureKey
+            val positiveValidator =
+              PositiveValidator::class.java.getDeclaredConstructor().newInstance()
+                as ValidatorAnnotation1<Any, FailureT?>
+            validator =
+              Validator<ValidatableT?, FailureT?> {
+                positiveValidator.validate(
+                  field,
+                  FieldUtils.readField(field, bean, true) as Any,
+                  map?.get(failureKey),
+                  none,
+                )
+              }
+            liftToEtr<FailureT, ValidatableT>(validator, none)
+          }
+          "MaxForInt" -> {
```

Comment:
This is a good first cut, I see there can be scope for refactoring
- Avoid some duplicate code
- If possible, break into small functions
- Mix of when and if-else (inside when case) can get confusing, you can convert all of them to `when`. Btw, you can convert if-else to when using IDE itself, keep ur cursor on `if` and option+enter gives you an option to convert